File: lib/models/user_profile.dart
/// 用户画像
class UserProfile {
  final String nickname;
  final double height; // cm
  final double weight; // kg
  final double bmi;
  final FitnessLevel fitnessLevel;
  final String scene;
  final int timeBudget; // 分钟
  final List<String> limitations;
  final String equipment;
  final String goal;
  final int weeklyDays;
  final List<String> preferredTime;

  UserProfile({
    required this.nickname,
    required this.height,
    required this.weight,
    required this.bmi,
    required this.fitnessLevel,
    required this.scene,
    required this.timeBudget,
    required this.limitations,
    required this.equipment,
    required this.goal,
    required this.weeklyDays,
    required this.preferredTime,
  });

  factory UserProfile.fromJson(Map<String, dynamic> json) {
    return UserProfile(
      nickname: json['nickname'] as String,
      height: (json['height'] as num).toDouble(),
      weight: (json['weight'] as num).toDouble(),
      bmi: (json['bmi'] as num).toDouble(),
      fitnessLevel: FitnessLevel.values.firstWhere(
        (e) => e.name == json['fitnessLevel'],
        orElse: () => FitnessLevel.beginner,
      ),
      scene: json['scene'] as String,
      timeBudget: json['timeBudget'] as int,
      limitations: List<String>.from(json['limitations'] as List),
      equipment: json['equipment'] as String,
      goal: json['goal'] as String,
      weeklyDays: json['weeklyDays'] as int,
      preferredTime: List<String>.from(json['preferredTime'] as List),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nickname': nickname,
      'height': height,
      'weight': weight,
      'bmi': bmi,
      'fitnessLevel': fitnessLevel.name,
      'scene': scene,
      'timeBudget': timeBudget,
      'limitations': limitations,
      'equipment': equipment,
      'goal': goal,
      'weeklyDays': weeklyDays,
      'preferredTime': preferredTime,
    };
  }

  static double calculateBMI(double height, double weight) {
    return weight / ((height / 100) * (height / 100));
  }
}

enum FitnessLevel {
  beginner,    // 零基础
  occasional,  // 偶尔运动
  regular,     // 规律运动
}

extension FitnessLevelExtension on FitnessLevel {
  String get label {
    switch (this) {
      case FitnessLevel.beginner:
        return '零基础';
      case FitnessLevel.occasional:
        return '偶尔运动';
      case FitnessLevel.regular:
        return '规律运动';
    }
  }
}

File: backend/app/models/user_profile.py
"""用户画像 ORM 模型"""
import uuid
from datetime import datetime
from sqlalchemy import Column, String, Numeric, Integer, DateTime, JSON, ForeignKey
from sqlalchemy.dialects.mysql import CHAR
from sqlalchemy.orm import relationship

from app.db.base import Base


class UserProfile(Base):
    """用户画像表"""

    __tablename__ = "user_profiles"

    id = Column(CHAR(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = Column(CHAR(36), ForeignKey("users.id", ondelete="CASCADE"), unique=True, nullable=False, index=True)

    # 基本信息
    nickname = Column(String(50), nullable=False)
    height = Column(Numeric(5, 2), nullable=False)  # cm
    weight = Column(Numeric(5, 2), nullable=False)  # kg
    bmi = Column(Numeric(4, 1), nullable=False)

    # 健身信息
    fitness_level = Column(String(20), nullable=False)  # beginner/occasional/regular
    scene = Column(String(20), nullable=False)  # bed/office/living/outdoor/hotel
    time_budget = Column(Integer, default=12, nullable=False)  # 分钟

    # 限制和装备
    limitations = Column(JSON, nullable=False, default=list)  # ["waist", "knee", ...]
    equipment = Column(String(20), nullable=False)  # none/mat/chair

    # 目标
    goal = Column(String(20), nullable=False)  # fat-loss/sedentary/strength/sleep
    weekly_days = Column(Integer, default=3, nullable=False)  # 2-7
    preferred_time = Column(JSON, nullable=False, default=list)  # ["morning", "noon", "evening"]

    # 时间戳
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    # 关联用户
    user = relationship("User", back_populates="profile")

    def __repr__(self):
        return f"<UserProfile(id={self.id}, user_id={self.user_id}, nickname={self.nickname})>"
